storagemarket/impl/provider_environments_test.go

```diff
@@ -0,0 +1,67 @@
+package storageimpl
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/filecoin-project/go-fil-markets/shared_testutil"
+	"github.com/filecoin-project/go-fil-markets/storagemarket"
+	"github.com/filecoin-project/go-fil-markets/storagemarket/impl/clientutils"
+	"github.com/ipfs/go-cid"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGeneratePieceCommitment(t *testing.T) {
+	carV1Path := filepath.Join("storagemarket", "fixtures", "test.car")
+	_, carv2 := shared_testutil.GenCARV2(t, carV1Path)
+	defer os.Remove(carv2)
+
+	env := &providerDealEnvironment{}
+
+	pieceCid, _, err := env.GeneratePieceCommitment(cid.Cid{}, carv2)
+	require.NoError(t, err)
+	require.NotEmpty(t, pieceCid.String())
+
+	// generate CommP again = should get back the same CommP
+	pieceCid2, _, err := env.GeneratePieceCommitment(cid.Cid{}, carv2)
+	require.NoError(t, err)
+	require.NotEmpty(t, pieceCid2.String())
+	require.Equal(t, pieceCid, pieceCid2)
+
+	// TODO Generate Commp for a different CAR file -> should get a different commP.
```

Comment:
TODO Write test to Generate Commp for a different CAR file -> should get a different commP.